README.md
# exBlog

exBlog is a free and open source blogging platform based on PHP and MySQL, which supported Multisites feature.

All source code are originally created in GB2312 (Simplified Chinese) character set and then converted to UTF-8 character set.

# Version

The version of master branch is 1.5.1. You may switch to other versions in the branches.

- [2004-03-30]exBlogMix v1.1.0
- [2004-09-06] exBlog 1.2.0 beta 0909
- [2004-10-23] exBlog 1.2.0 RC1
- [2004-12-21] exBlog 1.2.0 L (Christmas Version)
- [2005-04-13] exBlog 1.3.0 beta2
- [2005-04-16] exBlog 1.3.0 Build 0416
- [2005-05-xx] exBlog 1.3.0 final
- [2005-05-02] exBlog 1.3.1 Build 0502
- [2005-09-07] exBlog 1.5.0 Build 200509071201
- [2005-12-11] exBlog 1.5.1 Build 200512111540

# Licenses
Version 1.1.0 exBlogMix License

Version 1.2.0 exBlog License

Version 1.3.0 exBlog License

Version 1.5.1 GNU General Public License


HISTORY.TXT
                                  HISTORY
                                ~~~~~~~~~~~~~
                      Current Version: exBlog ver:1.3.1
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

                      -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
                       Welcome to the exBlog's History!
                      -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

 exBlog Ver: 1.3.1	2005-05-02
 ^^^^^^^^^^^^^^^^^ 
   + [修复] 所有链接无效等全局变量所引发的一系列问题。
   + [修复] 某些空间的HTTP-405错误
   + [修复] 一些原有模板对 exBlog 1.3.x 的支持
   + [修复] 某些统计为零。
   + [修复] 文章空格缩进

   + [增加] 基本的禁止对文章评论功能
 
 exBlog Ver: 1.3.0	2005-04-16
 ^^^^^^^^^^^^^^^^^ 
   + [调整] 安装文件的人性化！HOT!
   + [调整] 模板全面调整！HOT!

   + [修复] 发帖 editblog.php 的时间部分做了调整，增加了日历直接选定。
   + [修复] 通过ID直接查找文章的 input框 原本回车进入空白页面，现解决
   + [修复] 完美的搜索关键词加亮
   + [修复] 文章隐藏
   + [修复] RSS2.0的修复
   + [修复] 某些空间的HTTP-405错误

   + [增加] 多语言支持搞定！HOT!
   + [增加] 伪静态地址搞定！HOT!
	请测试 http://www.yoursite.com/exurl.php/links.html 
	是否与 http://www.yoursite.com/index.php?play=links 页面相同 
	相同，则表示你的空间支持，可以使用 exurl 模板体验伪静态地址！
   + [增加] Trackback 搞定！HOT!
   + [增加] 可用地址后跟 tm 参数来前台改变模板
	用法：http://www.yoursite.com/index.php?tm=模板名
   + [增加] 验证码开关
   + [增加] 加入“隐藏媒体超级ubb”呵呵。支持很多格式。
   + [增加] QQ友好页面。可应用于QQ自定义面板的窄幅页面。
   + [增加] 站点关键词，注意不是自动加亮，而是页面头关键词
   + [增加] 文章关键词，注意不是自动加亮，而是页面头关键词
   + [增加] 文章摘要
   + [增加] 控制每页显示文章篇数,自动摘要截取字数等设置
   + [增加] 强化后台文章编辑列表多条件自定义查询
   + [增加] 批量移动文章及其评论
   + [增加] 更多后台设置
   + [增加] 文件上传管理
   + [增加] 上传文件简单防盗链
   + [增加] 强化后台 评论管理 编辑列表多条件自定义查询
   + [增加] 词汇解释系统


 exBlog Ver: 1.2.0 [L] 圣诞版	2004-12-21
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

   + [修复] iMSN 模版的一个公告显示错误
   + [修复] 后台的页面标题错误
   + [修复] 各模版中的版本号错误
   + [修复] "编辑分类"的 Bug
   + [修复] 公告附件上传成功后不显示的Bug
   + [修复] 公告附件部分文件类型上传失败的Bug
   + [修复] i7c模版的一个图片错误
   + [修复] 日历的显示错误(图片红叉, 窄版变宽版等等)
   + [修复] pink模板的表格显示错误, 改为自动判断单元格高度
   + [修复] 无评论或者无友情连接时显示 NULL 的 Bug, 改为显示"当前无任何评论"和"当前无任何友情连接"
   + [修复] 完善 UBB 系统, 为所有标签添加了按钮以及对应的 js 代码, 方便用户使用. 更强大的“万能标签”功能将在下一个版本 1.3.0 放出
   + [修复] 应广大网友要求, 恢复后台的简单风格, 细微调整了后台的一些连接排列
   + [修复] 修复了备份, 导出数据库的安全漏洞
   + [修复] 修复"编辑公告"白屏

   + [增加] 全面启用新的版本号规则: 版本号最后一位为偶数的是稳定版, 奇数为测试版, [L] 为精简版, [D] 为豪华版 
   + [增加] 禁止访客注册与管理员同名的"个性评论ID"
   + [增加] 发表新文章时就应该允许编辑发布时间, 而不仅仅是等到发布之后再来编辑
   + [增加] 圣诞模板 xmas, 此模版是严格按照新的 html 规范以及 CSS 2.0 编写的, 同时也兼容了老规范, 所以可能在 IE 6.0 以下版本浏览器会出现显示故障, 故未作为默认模版, 有需要的用户可以自行选择。
   + [增加] Blog 文章批量删除功能
   + [增加] 文章"提交之前"使用JavaScript检查"文章标题"以及"正文"是否为空, 如果为空警告, 进一步防止疏忽导致文章内容丢失的问题, 虽然会把正文自动保存到剪贴板, 但是双保险更安全嘛. :)
   + [增加] 部分 UBB 自动识别
   + [增加] "最新10篇回复"RSS
   + [增加] 删除 Blog 时警告信息, 最大程度的防止误删除
   + [增加] 公告的发布能力, 允许公告中使用各种UBB标签以及上传附件
   + [增加] 天气中的"阴天"选项
   + [增加] 发表Blog以及公告时输入框的拉长, 缩短功能
   + [增加] 在不允许访问的目录下放置了空的index.html文件以增加安全性
   
   + [取消] 无任何Blog文章时的评论以及阅读全文的连接, 修复逻辑错误
   + [取消] 因时间仓促, 加上功能不成熟, 暂时去掉了 html 编辑器功能, 今后会在适当的时候重新放出
   
   + [网站] 从下一个版本开始我们将只提供 i7c, pink 两套模版, 更多模版请关注我们的模版网站 http://skin.exblog.net/
   + [网站] 从下一个版本开始我们将只发布 Blog 最基本的功能模块, 而一些增强插件如 RSS 0.91 支持等请关注我们的插件网站 http://plugin.exblog.net/
   
   + [人事] 热烈欢迎 Anthrax 加入开发团队 (鼓掌! 鼓掌!)

   + [特别] 圣诞临近, 让我们的 exBlog 1.2.0 [L] 圣诞版 作为我们为大家准备的一份小小的圣诞礼物吧, 祝大家圣诞快乐. 希望大家在即将到来的新的一年里学习进步, 工作顺利.
            考试临近, exBlog 的开发暂停至 2005年01月15日, 下一个版本 1.3.0 将会给您带来更多惊喜.



 exBlog Ver: 1.2.0 RC3	2004-11-27
 ^^^^^^^^^^^^^^^^^^^^^ 

   + 所有的图标(RSS1.0, RSS2.0, ATOM, CHANNEL, MYSQL, PHP, EXBLOG 共七个)。
   + 修改完所有的模板, 完全适合RC3的要求
   + RSS,ATOM的Bugs的修正(现在可以正确导入了)。如：blog发布时间等
   + CHANNEL频道功能, 请确保“upload”目录可写(非Windows类主机请将upload目录属性设置为777, Windows类主机就不用动了:))。
   + 修改点添加/编辑公告, 提示语法错误, 屏幕左下脚有黄色图标表示出错
   + 所有标签都应类似[url]支持选中对象后自动用标签夹住
   + 修改 i7c模板的RSS2.0在浏览器中点击白屏
   + 解决了优化, 备份数据库是对整个库操作, 有安全性隐患
   + 解决了非图片文件, 如exe等其他文件类型无法上传
   + 点添加/编辑公告, 提示语法错误, 屏幕左下脚有黄色图标表示出错
   + 如果当前无评论或者无友情连接, 不应该显示NULL而应该显示“当前无XXXXXXX”
   + 手机浏览最新几篇日记
   + 手机发表日记,有用户名认证,有分类选择,置顶选择,天气选择,隐藏选择,全面兼容WEB的blog发表体系.
   + 手机浏览网站信息, 包括共有blog数目, 评论数目, 今日访问, 总共访问
   + 手机浏览站长信息, 包括QQ,E-mail,name,age等信息, 以上信息自动从现有数据库表中读取, 也可将自己的手机号手动输入显示文件
   + 手机浏览评论,发表评论
   + 手机号码自动与用户名绑定, 可根据设定号码自动判断blog用户, 省去登陆
   + 后台编辑修改编辑首页为查看首页, 可以在不退出的情况下直接在右边显示效果, 省去不满意再次登陆的麻烦。
   + 广泛支持常见ubb。media 标签自动识别媒体文件。想法：一般的ubb我们支持, 但较特别的我们并不提倡使用。
   + 所见即所得Html编辑器。让HTMLBlog和UBBBlog共存。(评论, 公告是否应该支持Html, 还有待商榷)
	另：	HTMLBlog内的UBB代码不会起作用。
		html状态同样支持原来的表情代码哦! 
		可选用 fckeditor 2.0 Beta2 插件, 替换配套Html编辑器, 但暂时并不推荐。

 exBlog Ver: 1.2.0 RC2	2004-10-30
 ^^^^^^^^^^^^^^^^^^^^^ 
   
   + 增加文件上传功能,和原来图片上传使用同一程式自动判断上传文件类型, 改进上传文件命名方式, 使用原始文件名命名, 便于管理
   + 增加文章隐藏功能 
   + 增加发表公告还原功能
   + 增加留言编辑, 删除, 站长回复功能
   + 表情图片添加方式改为在光标处添加
   + UBB 代码新增表格转换功能 (可选补丁)
   + 增加初级 WAP 浏览功能 (Hoooooooooooooooot!!!)
   + 增加数据库备份, 数据库优化功能
   + 增加 RSS 2.0, ATOM 0.3 聚合格式支持
   + 修复, 增加 coolsee, iMSN, madcn 三个新模板
   + 修复原程式UBB代码[rimg][limg]失灵的BUG 
   + 修复添加BLOG天气有夜晚选项, 如果选择造成数据库查询失败的BUG 
   + 修复编辑公告失灵BUG
   + 修复查看评论时再点击评论链接错误的BUG 
   + 编辑文章时, “是否置顶”以前的状态也应该从数据库中取出
   + 修复没有文章时天气图片显示失败的BUG)包括按月查询和按分类查询)
   + 修复增加分类, 友情链接时ID号从10000开始的BUG
   + 修复后台管理左框架JAVA失灵BUG
   + 提供“原始的简洁后台”补丁包! (可选补丁, 另听取网友意见, 如果喜欢Dvbbs后台风格的请在论坛上提出)
   + 增加了 madcn 模板并将原先的模板整合进来, 现在有五套模板了;)
   + 修正了新版UBB编辑器的一些小问题, 实现了表情/UBB代码随处插入
   + 修正最新评论处会显示UBB代码的BUG
   + 部分细节小修补 (Viva: 555..快累死了..)


 exBlog Ver: 1.2.0 RC1	2004-10-23
 ^^^^^^^^^^^^^^^^^^^^^
   
   + 增加了图片上传功能, 可以在后台进行图片上传设定, 选择是否给图片加水印, 水印的位置, 水印类型(文字或图片), 自动生成缩略图, 需要GD库支持, 请非Windows主机务必设置上传目录属性为777
   + 修改pink摸板, 使之能使用1.2.0所有新功能
   + 增加修改分类显示顺序功能
   + 增加修改友情链接顺序功能
   + 增加新版UBB编辑器, 不必手动编写代码, 可以选定添加, 在编辑BLOG, 编辑公告时都可使用! 如想把12345中的23两个字变红, 只要用鼠标选中23两字然后在选择工具栏上的红色即可
   + 增加发表BLOG时内容自动复制剪贴板功能, 可以一键恢复
   + UBB代码嵌套BUG修复, UBB支持更加完美
   + 修正发表评论UBB代码失灵BUG
   + 修正天气状况设定, 单独设置表格存放
   + 修正已通过友情链接仍然显示拒绝链接的BUG
   + 修正按日期查询失败BUG
   + 修正了编辑以往BLOG时天气需要重新设定的问题, 本版本会自动独取原来的天气设定
   + 修正按日期查询天气图片显示错误BUG
   + 增加修改BLOG日期功能
   + 修正公告编辑显示的是公告删除的BUG
   + 取消了exBook的捆绑
   + 前, 后台一些细微的变化及bugs修复
   + sheer, netxiong, madcn 正式加入 exSoft团队

RC2 版预告: 
   + RSS 2.0加入
   + 初步支持WAP接入
   + 跟IPB论坛无缝接入
   + 新skin和iMSN skin修复

 exBlog Ver: 1.2.0 PF1	2004-09-06
 ^^^^^^^^^^^^^^^^^^^^^
   + 增加后台登录验证码功能.
   + 增加网友评论注册功能.
   + 增加了统计在线人数功能.
   + 增加了天气表情风格功能.
   + 增加了 blog 置顶功能.
   + 增加了前台可申请友情链接功能.

admin/main.php
<?php
/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 *
 *  File: main.php
 *  Author: feeling <feeling@exblog.org>
 *  Date: 2005-06-11 15:02
 *  Homepage: www.exblog.net
 *
 * $Id: main.php,v 1.1.1.1 2005/07/01 09:21:20 feeling Exp $
 * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */

require_once("./public.php");

class AdminHome extends CommonLib
{
	function AdminHome()
	{
		// 装入数据库配置并实例化
		$this->_loadDatabaseConfig();
		// 装入系统配置
		$this->_loadSystemOptions();
		// 装入语言文件
		include_once("../{$this->_config['global']['langURL']}/public.php");
		$this->_lang['public'] = $langpublic;
		$this->_lang['public'] = array_merge($this->_lang['public'], $langpubfun);
		// 装入功能对应语言文件
		include_once("../{$this->_config['global']['langURL']}/main.php");
		$this->_lang['main'] = $lang;
		include_once("../{$this->_config['global']['langURL']}/blogadmin.php");
		$this->_lang['admin'] = $langadfun;

		@session_start();
		// 用户验证
//		if (FALSE == $this->checkUser(0)) $this->_showError($this->_lang['public'][21], "./login.php");

		if (FALSE == empty($_GET['action']) && 'delInstall' == $_GET['action']) $this->_delInstallFile();
		else $this->_printOutPage();

		exit();
	}

	function _printOutPage()
	{
		$this->_getServerInfo();
		$this->_listTeamMembers();
		$this->printPageHeader();
		echo "<table cellpadding=4 cellspacing=1 border=0 width=100% align=center class=border><tr bgcolor=#99CC66><td bgcolor=#FFFFFF class=main><br>";

		if (FALSE == $this->_checkInstallFile()) echo "{$this->_lang['admin'][14]}{$this->_lang['admin'][15]}{$this->_lang['admin'][16]}";

		echo "<br><table width=100% border=0 cellpadding=1 cellspacing=1 class=main><tr><td colspan=2 class=menu align=center><b>{$this->_lang['main'][0]}</b></td></tr><tr><td height=23 colspan=2><b>- {$this->_lang['main'][1]}</b></td></tr><tr><td width=20% height=23></td><td width=80%>{$this->_lang['main'][2]}- {$this->status['num_blog']} {$this->_lang['main'][3]}- {$this->status['num_comment']} {$this->_lang['main'][4]}- {$this->status['num_trackback']} {$this->_lang['main'][1]}{$this->_lang['main'][15]}- {$this->status['num_visited']} <br /> {$this->_lang['main'][5]}- {$this->status['db_size']} {$this->_lang['main'][6]}- {$this->status['uptime']} {$this->_lang['main'][7]}</td></tr><tr><td height=23 colspan=2><b>- {$this->_lang['main'][8]}</b></td></tr><tr><td width=20% height=23 align=center>{$this->_lang['main'][9]}</td><td width=80%>&nbsp;{$this->status['www']['server']}</td></tr><tr><td width=20% height=23 align=center>{$this->_lang['main'][10]}</td><td width=80%>&nbsp;{$this->status['database']['version']}</td></tr><tr><td width=20% height=23 align=center>{$this->_lang['main'][25]}</td><td width=80%>&nbsp;{$this->status['database']['host']}</td></tr><tr><td width=20% height=23 align=center>{$this->_lang['main'][26]}</td><td width=80%>&nbsp;{$this->status['database']['client']}</td></tr><tr><td height=23 colspan=2><b>- {$this->_lang['main'][27]}</b></td></tr><tr><td width=20% height=23 align=center>{$this->_lang['main'][28]}</td><td width=80%>&nbsp;{$_SESSION['exPass']}</td></tr><tr><td width=20% height=23 align=center>{$this->_lang['main'][29]}</td><td width=80%>&nbsp;" . date("Y-m-d H:i:s", $_SESSION['exUserlastvisit']) . "</td></tr><tr><td width=20% height=23 align=center>{$this->_lang['main'][30]}</td><td width=80%>&nbsp;{$_SESSION['exUserIpAddress']}</td></tr></table><hr size=0 noshade width=90%>";

		echo "<table cellpadding=4 cellspacing=1 border=0 width=95% align=center class=main><tr><td class=menu colspan=2 height=25 align=center><b>{$this->_lang['main'][11]}</b></td></tr>";

		echo "<tr><td width=25% height=23 nowrap><b>- {$this->_lang['main'][16]}</b></td><td width=75% height=23>";
		$this->_printMemberList($this->members['manager']);
		echo "</td></tr>";

		echo "<tr><td width=25% height=23 nowrap><b>- {$this->_lang['main'][17]}</b></td><td width=75% height=23>";
		$this->_printMemberList($this->members['coder']);
		echo "</td></tr>";

		echo "<tr><td width=25% height=23 nowrap><b>- {$this->_lang['main'][18]}</b></td><td width=75% height=23>";
		$this->_printMemberList($this->members['skiner']);
		echo "</td></tr>";

		echo "<tr><td width=25% height=23 nowrap><b>- {$this->_lang['main'][19]}</b></td><td width=75% height=23>";
		$this->_printMemberList($this->members['documenter']);
		echo "</td></tr>";

		echo "<tr><td width=25% height=23 nowrap><b>- {$this->_lang['main'][20]}</b></td><td width=75% height=23>";
		$this->_printMemberList($this->members['tester']);
		echo "</td></tr>";

		echo "<tr><td width=25% height=23 nowrap><b>- {$this->_lang['main'][21]}</b></td><td width=75% height=23>";
		$this->_printMemberList($this->members['benefactor']);
		echo "</td></tr>";

		echo "</table><hr size=0 noshade width=90%>";

		echo "<table width=100% border=0 cellpadding=1 cellspacing=1 class=main><tr><td class=menu colspan=2 height=25 align=center><b>{$this->_lang['main'][12]}</b></td></tr><tr><td height=23 colspan=2>{$this->_lang['main'][13]}</td></tr><tr><td width=20% height=23 align=center>{$this->_lang['main'][22]} :</td><td width=80%>&nbsp; <a href=http://www.exBlog.net/>http://www.exBlog.net/</a></td></tr><tr><td width=20% height=23 align=center>{$this->_lang['main'][23]} :</td><td width=80%>&nbsp; <a href=http://bbs.exblog.net/>{$this->_lang['main'][14]}</a></td></tr><tr><td width=20% height=23 align=center>{$this->_lang['main'][24]} :</td><td width=80%>&nbsp; <a href=mailto:support@exblog.net>support@exblog.net</a></td></tr></table></td></tr></table>";

		$this->printPageFooter();
	}

	function _checkInstallFile()
	{
		$install_file = "./install.php";
		if (TRUE == @file_exists($install_file)) return FALSE;
		else return TRUE;
	}

	function _printMemberList($members)
	{
		$tpl_with_email = " <a href=mailto:%s>%s</a>,";
		$tpl_without_email = " %s,";
		$tpl_with_homepage = " <a href=http://%s>%s</a>,";
		for ($i = 0; $i < count($members); $i++)
		{
			if (TRUE == isset($members[$i]['email']) && FALSE == empty($members[$i]['email'])) printf($tpl_with_email, $members[$i]['email'], $members[$i]['name']);
			elseif (TRUE == isset($members[$i]['homepage']) && FALSE == empty($members[$i]['homepage'])) printf($tpl_with_homepage, $members[$i]['homepage'], $members[$i]['name']);
			else printf($tpl_without_email, $members[$i]['name']);
		}
	}

	function _getServerInfo()
	{
		$this->status['www']['server'] = getenv("SERVER_SOFTWARE");

		$query_string = "SELECT VERSION() AS version";
		$this->_dbop->query($query_string);
		$tmp = $this->_dbop->fetchArray($this->_dbop->query_id, 'ASSOC');
		$this->_dbop->freeResult();
		$this->status['database']['version'] = $tmp['version'];
		$this->status['database']['host'] = mysql_get_host_info();
		$this->status['database']['client'] = mysql_get_client_info();

		$query_string = "select blogCount, commentCount, trackbackCount from {$this->_dbop->prefix}global limit 0, 1";
		$this->_dbop->query($query_string);
		$tmp = $this->_dbop->fetchArray($this->_dbop->query_id, MYSQL_ASSOC);
		$this->_dbop->freeResult();
		$this->status['num_blog'] = $tmp['blogCount'];
		$this->status['num_comment'] = $tmp['commentCount'];
		$this->status['num_trackback'] = $tmp['trackbackCount'];

		// 获取访问统计数
		$query_string = "select visits from {$this->_dbop->prefix}visits order by currentDate desc limit 0, 1";
		$this->_dbop->query($query_string);
		if (TRUE == $this->_dbop->getNumRows()) $tmp = $this->_dbop->fetchArray(0, 'ASSOC');
		else $tmp['visits'] = 0;
		$this->_dbop->freeResult();
		$this->status['num_visited'] = $tmp['visits'];

		$this->status['db_size'] = $this->_getDatabaseSize();

		$this->status['uptime'] = $this->_getUptime();
	}

	function _getDatabaseSize()
	{
		$size = 0;
		$tables = array('aboutme', 'admin', 'announce', 'blog', 'comment', 'global', 'links', 'sort', 'user', 'visits', 'tags');
		array_walk($tables, array($this, "_addTablePrefix"), $this->_dbop->prefix);
		$query_string = "SHOW TABLE STATUS FROM {$this->_dbop->dbname}";
		$this->_dbop->query($query_string);
		$tmp = $this->_dbop->fetchArrayBat($this->_dbop->query_id, MYSQL_ASSOC);
		$this->_dbop->freeResult();
		for ($i = 0; $i < count($tmp); $i++)
		{
			if (TRUE == in_array($tmp[$i]['Name'], $tables)) $size += $tmp[$i]['Data_length'] + $tmp[$i]['Index_length'] + $tmp[$i]['Data_free'];
		}
		return $this->_formatFileSize($size);
	}

	function _getUptime()
	{
		$now_time = time();
		$day = 1;
		if ($now_time != $this->_config['global']['initTime']) $day = ($now_time - $this->_config['global']['initTime']) / 3600 / 24;
		return ceil($day);
	}

	function _addTablePrefix(&$value, $key, $prefix)
	{
		$value = $prefix . $value;
	}

	function _listTeamMembers()
	{
		$this->members['manager'] = array(
			array('name' => 'netxiong', 'email' => 'netxiong@hotmail.com'),
			array('name' => 'Viva', 'email' => 'sachxp@126.com')
		);
		$this->members['coder'] = array(
			array('name' => 'feeling', 'email' => 'feeling_2005@126.com'),
			array('name' => 'elliott'),
			array('name' => 'Tomy'),
			array('name' => 'Sheer'),
			array('name' => '流水流云'),
			array('name' => 'Anthrax')
		);
		$this->members['skiner'] = array(
			array('name' => 'madman', 'homepage' => 'www.madcn.com'),
			array('name' => 'Hunter')
		);
		$this->members['documenter'] = array(
			array('name' => 'Viva', 'email' => 'sachxp@126.com'),
			array('name' => 'Tomy')
		);
		$this->members['tester'] = array(
			array('name' => 'BBS', 'homepage' => 'bbs.exBlog.net/')
		);
		$this->members['benefactor'] = array(
			array('name' => '混蛋70', 'homepage' => 'www.w3u.cn'),
			array('name' => 'codesky', 'homepage' => 'www.flydream.cn'),
			array('name' => 'neptune', 'homepage' => 'neptune.t35.com/blog/'),
			array('name' => 'aplapl', 'homepage' => 'aplapl.hwagain.com/myblog'),
			array('name' => 'ryhbgs'),
			array('name' => 'rick'),
			array('name' => 'gogo'),
			array('name' => 'howell', 'email' => 'vwell@163.com'),
			array('name' => '蓝鸟', 'email' => 'admin@wfdxh.com')
		);
	}

	function _delInstallFile()
	{
		$install_file = "./install.php";
		if (TRUE == file_exists($install_file)) $result = @unlink($install_file);
		if (TRUE == $result) $this->showMesg($this->_lang['admin'][17], $_SERVER['PHP_SELF']);
		else $this->_showError($this->_lang['admin'][14]);
	}
}

new AdminHome();
?>